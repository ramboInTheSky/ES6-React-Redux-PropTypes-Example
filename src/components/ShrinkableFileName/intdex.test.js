import React from 'react';
import { shallow } from 'enzyme';
import AttachmentsTableFileName from '.';

describe('<AttachmentsTableFileName />', () => {
  let el;

  beforeEach(() => {
    el = shallow(<AttachmentsTableFileName name={'Hello world'} />);
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should render correctly with a name', () => {
    expect(el).toMatchSnapshot();
  });

  it('should render a span for mobile and a span for desktop', () => {
    expect(el.find('span').length).toBe(2);
  });

  it('should shrink down the file names for desktop and for mobile', () => {
    el = shallow(
      <AttachmentsTableFileName
        name={'somecrazilybigfilenamethatshouldtriggertheshortenerfunction.jpg'}
      />
    );

    const desktopVersion = el.find('span').at(0);
    const mobileVersion = el.find('span').at(1);

    expect(desktopVersion.text()).toBe('somecrazilybigfilenamethat...ion.jpg');
    expect(mobileVersion.text()).toBe('somecra...ion.jpg');
  });
});
