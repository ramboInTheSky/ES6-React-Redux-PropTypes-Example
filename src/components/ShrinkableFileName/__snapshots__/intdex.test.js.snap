// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AttachmentsTableFileName /> should render correctly with a name 1`] = `
<styled.span>
  <span
    className="d-none d-lg-block"
  >
    Hello world
  </span>
  <span
    className="d-lg-none"
  >
    Hello world
  </span>
</styled.span>
`;
